docs/GeoReplication.md

```diff
@@ -0,0 +1,137 @@
+global namespace, and configures that namespace to replicate between two ore more
+of the provisioned clusters. Any message published on any topic in that namespace
+will then be replicated to all clusters in that set.
+
+Messages are first stored in the local cluster, and then forwarded
+asynchronously to the remote clusters.
+
+In normal case, with no connectivity problems, messages are replicated immediately,
+at the same time as they are dispatched to local consumers. Typically, end-to-end
+delivery latency is defined by the network *RTT* between the remote regions.
+
+Applications can create producers and consumers in each of the clusters,
+even when the remote clusters are not reachable (eg: during a network
+partition).
+
+Subscriptions are local to the clusters in which they are created and they
+cannot be transferred between clusters.
+
+![Replication Diagram](img/GeoReplication.png)
+
+In the above example, the topic is being replicated between 3 clusters,
+***Cluster-A***, ***Cluster-B*** and ***Cluster-C***.
+
+All messages produced in any cluster will be delivered to all subscriptions
+in all the clusters. In this case, both ***C1*** and ***C2*** will receive all
+messages published by ***P1***, ***P2*** and ***P3***. Ordering is still
+guaranteed per-producer.
+
+## Configuring replication
+
+### Granting permissions to properties
+
+When each tenant is on-boarded to a Pulsar cluster, it needs to be
+granted permissions to be able to create namespaces in any particular cluster.
+
+When creating a new property, specify all the intended clusters:
+```shell
+$ bin/pulsar-admin properties create my-property    \
+        --admin-roles my-admin-role                 \
+        --allowed-clusters us-west,us-east,us-cent
+```
+
+You can update the configuration of an existing property by using `update`
+instead of `create`.
+
+### Creating the namespace
+
+To use replication, we need a *global* topic, that is to say, a topic
+that belong to a *global* namespace, not tied to any particular cluster.
+
+Global namespaces are assigned to the `global` virtual cluster. To create a new
+namespace:
+
+```shell
+$ bin/pulsar admin namespaces create my-property/global/my-namespace
+```
+
+Initially, the namespace is not assigned to any cluster.
+
+```shell
+$ bin/pulsar admin namespaces set-clusters
+                          my-property/global/my-namespace \
+                          --clusters us-west,us-east,us-cent
+```
+
+The list of replication clusters can be changed at any time, with no
+disruption to ongoing traffic. Replication channels will be immediately
+set up or stopped in all the clusters as soon as the configuration changes.
+
+### Using global topics
+
+At this point you can just create a producer or consumers and automatically
+create a global topic.
+
+Typically, each application will use the `serviceUrl` for the local cluster.
+
+#### Selective replication
```

Comment:
By default, messages are replicated  to all  configured clusters for the namespace.  You can restrict replication selectively, by specifying a replication list for a message. The message will then be replicated only to the subset in the replication list